source/common/upstream/maglev_lb.cc

```diff
@@ -14,27 +14,48 @@ MaglevTable::MaglevTable(const HostVector& hosts, uint64_t table_size) : table_s
     return;
   }
 
+  // Compute maximum host weight. If this is zero, we are doing unweighted Maglev.
+  uint32_t max_host_weight = 0;
+  for (const auto& host : hosts) {
+    max_host_weight = std::max(host->weight(), max_host_weight);
+  }
+  // Integer range to normalize and scale backend weights to.
+  const uint32_t backend_weight_scale = 65535;
+
   // Implementation of pseudocode listing 1 in the paper (see header file for more info).
   std::vector<TableBuildEntry> table_build_entries;
   table_build_entries.reserve(hosts.size());
   for (const auto& host : hosts) {
     const std::string& address = host->address()->asString();
+    const uint32_t weight =
```

Comment:
What's the purpose of the scaling? Since the paper doesn't cover can you add more comments?